expyfun/stimuli/_tracker.py

```diff
@@ -0,0 +1,629 @@
+        return self._step_size_down[self._step_index()]
+
+    # =========================================================================
+    # Define all the public properties
+    # =========================================================================
+    @property
+    def up(self):
+        return self._up
+
+    @property
+    def down(self):
+        return self._down
+
+    @property
+    def step_size_up(self):
+        return self._step_size_up
+
+    @property
+    def step_size_down(self):
+        return self._step_size_down
+
+    @property
+    def stop_criterion(self):
+        return self._stop_criterion
+
+    @property
+    def stop_rule(self):
+        return self._stop_rule
+
+    @property
+    def start_value(self):
+        return self._start_value
+
+    @property
+    def x_min(self):
+        return self._x_min
+
+    @property
+    def x_max(self):
+        return self._x_max
+
+    @property
+    def stopped(self):
+        return self._stopped
+
+    @property
+    def x(self):
+        return self._x
+
+    @property
+    def x_current(self):
+        return self._x_current
+
+    @property
+    def responses(self):
+        return self._responses
+
+    @property
+    def n_trials(self):
+        return self._n_trials
+
+    @property
+    def n_reversals(self):
+        return self._n_reversals
+
+    @property
+    def reversals(self):
+        return self._reversals
+
+    @property
+    def reversal_inds(self):
+        return np.where(self._reversals)[0]
+
+    # =========================================================================
+    # Utility functions
+    # =========================================================================
+    def _test_prob(self, chance, thresh=None):
+        from scipy.stats import norm
+        if thresh is None:
+            thresh = self._start_value
+        slope = np.mean([self._step_size_up.mean(),
+                         self._step_size_down.mean()]) ** -1.
+        return lambda x: (chance + (1 - chance) *
+                          norm.cdf((x - thresh) * slope))
+
+    def _test_iprob(self, chance, thresh=None):
+        from scipy.stats import norm
+        if thresh is None:
+            thresh = self._start_value
+        slope = ((self._step_size_up[-1] +
+                  self._step_size_down[-1]) / 2) ** -1.
+        return lambda x: (thresh +
+                          norm.ppf((x - chance) / (1 - chance)) / slope)
+
+    # =========================================================================
+    # Display functions
+    # =========================================================================
+    def plot(self):
+        import matplotlib.pyplot as plt
+        line = plt.plot(1 + np.arange(self._n_trials), self._x, 'k.-')
+        dots = plt.plot(1 + np.where(self._reversals > 0)[0],
+                        self._x[self._reversals > 0], 'ro')
+        plt.show()
+        plt.xlabel('Trial number')
+        plt.ylabel('Level')
```

Comment:
It has always seemed like the member functions of axes objects are spotty. For example, `ax.xlabel(...)` doesn't work. Is `ax.xaxis.set_label('Foo')` the right way to do it?